docs/developer-guide.rst

```diff
@@ -0,0 +1,44 @@
+Developer Guide
+===============
+
+
+Pre-commit Checks
+-----------------
+
+This project is configured for use with `pre-commit <https://pre-commit.com>`_ to perform some static code analysis checks. By default, pre-commit behavior is disabled. To enable pre-commit in a local checkout, first install pre-commit (consider using `pipx <https://pipxproject.github.io/pipx/>`_), then install the hooks with ``pre-commit install``.
+
+For example::
+
+	tahoe-lafs $ pre-commit install
+	pre-commit installed at .git/hooks/pre-commit
+	tahoe-lafs $ echo "def foo():\n\t''" > src/allmydata/tabbed.py
```

Comment:
My version of `echo` prints a literal `\n\t` here rather than a newline and tab. I suggest suggesting `printf` instead.